src/Core.tsx

```diff
@@ -309,7 +309,11 @@ export class Core extends React.Component<IModalProps, IModalState> {
      const { providerController } = this.props
 
      const providerName = provider.name || 'Provider'
-
+     this.setState({
+       currentStep: 'loading',
+       loadingReason: i18next.t('Please open your wallet'),
+       selectedProviderUserOption: { ...selectedProviderUserOption, provider }
```

Comment:
Why is `selectedProviderUserOption` set here? It looks like it should only be set once the provider has connected in the `then` right below.

Reply:
Removed